tests/test_slow_retrieval_attack.py

```diff
@@ -68,68 +68,20 @@
 repo_tool.disable_console_log_messages()
 
 
-class TestSlowRetrievalAttack(unittest_toolbox.Modified_TestCase):
 
-  @classmethod
-  def setUpClass(cls):
+def set_up_slow_retrieval_tests(obj, repo_name, mode):
```

Comment:
this looks like a new SlowRetrievalBase class Setup() masquerading as normal function: `obj` really must be an instance of Modified_TestCase here, because you use `make_temp_directory()`?

Looks a bit fishy as it is.